idaes/generic_models/properties/core/eos/enrtl.py

```diff
@@ -436,7 +437,7 @@ def rule_tau_expr(b, i, j):
             else:
                 alpha = getattr(b, pname+"_alpha")
                 G = getattr(b, pname+"_G")
-                if G[i, j].expr != 1:
+                if str(G[i, j].expr) != "1.0":
```

Comment:
Why `str()` here and not `value()`?

Reply:
This is because we need to distinguish between the case where the value happens to be 1, and where the expression itself is 1. I.e., I am checking for the equivalent of `Expression(expr=1)`.

Reply:
I'm not too sure, but if  G[i, j] is an expression and just 1, the type should be <class 'pyomo.core.expr.numvalue.NumericConstant'>.  You could check the type maybe?  Not sure the best way. I agree str() seems a little weird.